.github/workflows/test-macOS.yml

```diff
@@ -12,6 +12,11 @@ on:
   # Triggers the workflow on pull request events but only for the main branch
   pull_request:
     branches: [main]
```

Comment:
test-macOS can not be skipped,  because the `codecov` relies on it.

Reply:
Can remove codecov when only change doc?

Reply:
The GitHub `required` mark is static, the only choice is removing the `required` condition of `codecov`:

![image](https://user-images.githubusercontent.com/7237365/217211806-f3c5875e-f89b-4922-9eb1-b9326448e0e8.png)